torch/csrc/stable/ops.h

```diff
@@ -68,6 +68,15 @@ inline Tensor pad(
   return Tensor(ret0);
 }
 
+// We expect this to be the stable version of the amax.default op
+// with identical semantics to the existing amax.default op.
+inline Tensor amax(Tensor& self, int64_t* dim, int dim_len, bool keepdim) {
```

Comment:
In the high level C++ API I think it would be ideal not to expose `dim_len` as a parameter

While we're figuring out what to do with `ArrayRef` I would propose the following:

Let's have two overloads of `amax`
- one that takes `int64_t` dim
- one that takes `std::vector<int64_t>` (as we do for pad)

In the future, when we figure out IntArrayRef we'll add another overload :)

cc @janeyx99 for thoughts as well